proto/viam/app/v1/end_user.proto

```diff
@@ -23,3 +26,13 @@ message IsLegalAcceptedResponse {
 message AcceptLegalRequest {}
 
 message AcceptLegalResponse {}
+
+message RegisterApplicationRequest {
+  string application_name = 1;
+  string application_id = 2;
```

Comment:
sorry I think maybe my comment was misplaced. I think this one should be marked `optional`, although thinking about it more, maybe we should just omit it entirely and force using an auto-generated application ID. The reason is that this is required to be globally unique, and then we don't have to handle customers using an application ID like "app" and wrapping the FusionAuth response to give a detailed exception.

Reply:
from the ticket description: 
> Our gRPC endpoint should have application name and redirect URIs should be required parameters. (i.e. don’t use FusionAuth’s built in feature of generating a random client ID)

was thinking this meant we wanted them to pass an application id from the parenthesis, may have misunderstood. can remove if thats not the case!